FLT/Deformations/RepresentationTheory/Basic.lean

```diff
@@ -0,0 +1,379 @@
+import FLT.Deformations.RepresentationTheory.AbsoluteGaloisGroup
+import FLT.Deformations.RepresentationTheory.Etale
+import Mathlib.LinearAlgebra.Charpoly.Basic
+import Mathlib.LinearAlgebra.Matrix.Unique
+import Mathlib.RingTheory.Bialgebra.TensorProduct
+import Mathlib.RingTheory.HopfAlgebra.Basic
+
+open NumberField
+
+universe uK
+
+variable {K : Type uK} {L : Type*} [Field K] [Field L]
+variable {A : Type*} [CommRing A] [TopologicalSpace A]
+variable {B : Type*} [CommRing B] [TopologicalSpace B]
+variable {M N : Type*} [AddCommGroup M] [Module A M] [AddCommGroup N] [Module A N]
+variable {n : Type*} [Fintype n] [DecidableEq n]
+
+variable [NumberField K] (v : IsDedekindDomain.HeightOneSpectrum (𝓞 K))
+
+local notation3 "Γ" K:max => Field.absoluteGaloisGroup K
+local notation3 K:max "ᵃˡᵍ" => AlgebraicClosure K
+local notation3 "𝔪" => IsLocalRing.maximalIdeal
+local notation3 "κ" => IsLocalRing.ResidueField
+local notation "Ω" K => IsDedekindDomain.HeightOneSpectrum (𝓞 K)
+local notation "Kᵥ" => IsDedekindDomain.HeightOneSpectrum.adicCompletion K v
+local notation "𝒪ᵥ" => IsDedekindDomain.HeightOneSpectrum.adicCompletionIntegers K v
+local notation "Frobᵥ" => Field.AbsoluteGaloisGroup.adicArithFrob v
+
+attribute [local instance 100000]
+  instAlgebraSubtypeMemValuationSubring_fLT IntermediateField.algebra'
+  Algebra.toSMul Subalgebra.toCommRing Algebra.toModule
+  Subalgebra.toRing Ring.toAddCommGroup AddCommGroup.toAddGroup
+  ValuationSubring.smulCommClass IntermediateField.toAlgebra
+  IntermediateField.smulCommClass_of_normal
+  mulSemiringActionIntegralClosure
+  Subalgebra.algebra
+  CommRing.toCommSemiring
+  Valued.toIsUniformAddGroup
```

Comment:
What's this all about? Is it needed?

Reply:
It once makes everything timeout but somehow this issue is no more.